src/main/java/com/example/postgresdemo/service/QuestionService.java

```diff
@@ -0,0 +1,66 @@
+package com.example.postgresdemo.service;
+
+import com.example.postgresdemo.exception.ResourceNotFoundException;
+import com.example.postgresdemo.model.Question;
+import com.example.postgresdemo.model.QuestionRequestDTO;
+import com.example.postgresdemo.model.QuestionResponseDTO;
+import com.example.postgresdemo.repository.QuestionRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+@Service
+public class QuestionService {
+    private final QuestionRepository questionRepository;
+
+    public QuestionService(QuestionRepository questionRepository) {
+        this.questionRepository = questionRepository;
+    }
+
+    public Page<QuestionResponseDTO> findAll(Pageable pageable) {
+        return questionRepository.findAll(pageable)
+                .map(this::toQuestionResponseDTO);
+    }
+
+    public QuestionResponseDTO create(QuestionRequestDTO questionRequest) {
+        Question question = toQuestion(questionRequest);
+        return toQuestionResponseDTO(questionRepository.save(question));
+    }
+
+    public QuestionResponseDTO update(Long questionId, QuestionRequestDTO questionRequest) {
+        Question question = toQuestion(questionRequest);
+        return questionRepository.findById(questionId)
+                .map(foundQuestion -> {
+                    foundQuestion.setTitle(question.getTitle());
+                    foundQuestion.setDescription(question.getDescription());
+                    return toQuestionResponseDTO(questionRepository.save(foundQuestion));
+                }).orElseThrow(() -> new ResourceNotFoundException("Question not found with id " + questionId));
+
+    }
+
+    public void delete(Long questionId) {
```

Comment:
`.map()` seems unnecessary here, try something like this:
```suggestion
    public void delete(Long questionId) {
        Question question = questionRepository.findById(questionId)
            .orElseThrow(() -> new ResourceNotFoundException("Question not found with id " + questionId));
         questionRepository.delete(question);
    }
```